WFInfo/Ocr.cs

```diff
@@ -2153,34 +2148,38 @@ public static async Task updateEngineAsync()
             };
         }
 
-        public static bool VerifyWarframe()
-        {
-    
-            if (Warframe != null && !Warframe.HasExited)
-            { // don't update status
+        public static bool VerifyWarframe() {
+            if (Warframe != null && !Warframe.HasExited) { // don't update status
                 return true;
             }
-            Task.Run(() =>
-            {
-            foreach (Process process in Process.GetProcesses())
-                if (process.ProcessName == "Warframe.x64")
-                {
-                    if (process.MainWindowTitle == "Warframe")
-                    {
-                        HandleRef = new HandleRef(process, process.MainWindowHandle);
-                        Warframe = process;
-                        if (Main.dataBase.GetSocketAliveStatus())
-                            Debug.WriteLine("Socket was open in verify warframe");
-                        Task.Run(async () =>
-                        {
-                            await Main.dataBase.SetWebsocketStatus("in game");
-                        });
-                        Main.AddLog("Found Warframe Process: ID - " + process.Id + ", MainTitle - " + process.MainWindowTitle + ", Process Name - " + process.ProcessName);
-                        return true;
+            Task.Run(() => {
+                foreach (Process process in Process.GetProcesses())
+                    if (process.ProcessName == "Warframe.x64") {
+                        if (process.MainWindowTitle == "Warframe") {
+                            HandleRef = new HandleRef(process, process.MainWindowHandle);
+
+                            Warframe = process;
+                            if (Main.dataBase.GetSocketAliveStatus())
+                                Debug.WriteLine("Socket was open in verify warframe");
+                            Task.Run(async () =>
+                            {
+                                await Main.dataBase.SetWebsocketStatus("in game");
+                            });
+                            Main.AddLog("Found Warframe Process: ID - " + process.Id + ", MainTitle - " + process.MainWindowTitle + ", Process Name - " + process.ProcessName);
+
+                            //try and catch any UAC related issues
+                            try {
+                                bool _ = Warframe.HasExited;
+                                return true;
+                            }
+                            catch (Exception e) {
```

Comment:
Can we get more specific with exception type as this seems too much catchall?

Reply:
Could do, but I put it around just the method where it checks for UAC so I went on a leg that it would've already specified the error down to access violation rather than inability to find.

Reply:
In this case there's basically nothing else that could go wrong, but it wouldn't hurt to be specific with the exception type to be on the safe side and not risk future problems

Reply:
Added the specific exception we're interested in